sonar-ruby-plugin/src/test/java/org/sonarsource/ruby/externalreport/rubocop/RuboCopSensorTest.java

```diff
@@ -0,0 +1,242 @@
+import org.sonar.api.rules.RuleType;
+import org.sonar.api.utils.Version;
+import org.sonar.api.utils.log.LogTester;
+import org.sonar.api.utils.log.LoggerLevel;
+import org.sonarsource.ruby.plugin.RubyPlugin;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class RuboCopSensorTest {
+
+  private static final Path PROJECT_DIR = Paths.get("src", "test", "resources", "externalreport", "rubocop");
+
+  private static RuboCopSensor ruboCopSensor = new RuboCopSensor();
+
+  @Rule
+  public LogTester logTester = new LogTester();
+
+  @Test
+  public void test_descriptor() {
+    DefaultSensorDescriptor sensorDescriptor = new DefaultSensorDescriptor();
+    ruboCopSensor.describe(sensorDescriptor);
+    assertThat(sensorDescriptor.name()).isEqualTo("Import of RuboCop issues");
+    assertThat(sensorDescriptor.languages()).isEmpty();
+    assertNoErrorWarnDebugLogs(logTester);
+  }
+
+  @Test
+  public void no_issues_with_sonarqube_71() throws IOException {
+    List<ExternalIssue> externalIssues = executeSensorImporting(7, 1, "rubocop-report.json");
+    assertThat(externalIssues).isEmpty();
+    assertThat(logTester.logs(LoggerLevel.ERROR)).containsExactly("Import of external issues requires SonarQube 7.2 or greater.");
+  }
+
+  @Test
+  public void issues_with_sonarqube_72() throws IOException {
+    List<ExternalIssue> externalIssues = executeSensorImporting(7, 2, "rubocop-report.json");
+    assertThat(externalIssues).hasSize(3);
+
+    ExternalIssue first = externalIssues.get(0);
+    assertThat(first.primaryLocation().inputComponent().key()).isEqualTo("rubocop-project:useless-assignment.rb");
+    assertThat(first.ruleKey().toString()).isEqualTo("rubocop:Lint/UselessAssignment");
+    assertThat(first.type()).isEqualTo(RuleType.CODE_SMELL);
+    assertThat(first.severity()).isEqualTo(Severity.MAJOR);
+    assertThat(first.primaryLocation().message()).isEqualTo("Lint/UselessAssignment: Useless assignment to variable - `param`.");
+    TextRange firstRange = first.primaryLocation().textRange();
+    assertThat(firstRange.start().line()).isEqualTo(3);
+    assertThat(firstRange.start().lineOffset()).isEqualTo(2);
+    assertThat(firstRange.end().line()).isEqualTo(3);
+    assertThat(firstRange.end().lineOffset()).isEqualTo(7);
+
+    ExternalIssue second = externalIssues.get(1);
+    assertThat(second.primaryLocation().inputComponent().key()).isEqualTo("rubocop-project:yaml-issue.rb");
+    assertThat(second.ruleKey().toString()).isEqualTo("rubocop:Security/YAMLLoad");
+    assertThat(second.type()).isEqualTo(RuleType.VULNERABILITY);
+    assertThat(second.severity()).isEqualTo(Severity.MAJOR);
+    assertThat(second.primaryLocation().message()).isEqualTo("Security/YAMLLoad: Prefer using `YAML.safe_load` over `YAML.load`.");
+    TextRange secondRange = second.primaryLocation().textRange();
+    assertThat(secondRange.start().line()).isEqualTo(2);
```

Comment:
The assertions on text ranges would be more readable if we able to write something like 
```
assertThat(secondRange).isEqualTo(2, 7, 2, 11);
```

Reply:
changed to the following format `from line 2 offset 7 to line 2 offset 11`